Code/Framework/AzFramework/Platform/Linux/AzFramework/Windowing/NativeWindow_Linux_xcb.cpp

```diff
@@ -0,0 +1,243 @@
+        m_height = geometry.m_height;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::Activate()
+    {
+        LinuxXcbEventHandlerBus::Handler::BusConnect();
+
+        if (!m_activated) // nothing to do if window was already activated
+        {
+            m_activated = true;
+
+            xcb_map_window(m_xcbConnection, m_xcbWindow);
+            xcb_flush(m_xcbConnection);
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::Deactivate()
+    {
+        if (m_activated) // nothing to do if window was already deactivated
+        {
+            m_activated = false;
+
+            WindowNotificationBus::Event(reinterpret_cast<NativeWindowHandle>(m_xcbWindow), &WindowNotificationBus::Events::OnWindowClosed);
+
+            xcb_unmap_window(m_xcbConnection, m_xcbWindow);
+            xcb_flush(m_xcbConnection);
+        }
+        LinuxXcbEventHandlerBus::Handler::BusDisconnect();
+    }    
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    NativeWindowHandle NativeWindowImpl_Linux_xcb::GetWindowHandle() const
+    {
+        return reinterpret_cast<NativeWindowHandle>(m_xcbWindow);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::SetWindowTitle(const AZStd::string& title)
+    {
+        xcb_void_cookie_t xcbCheckResult;
+        xcbCheckResult = xcb_change_property(m_xcbConnection,
+                                               XCB_PROP_MODE_REPLACE,
+                                               m_xcbWindow,
+                                               XCB_ATOM_WM_NAME,
+                                               XCB_ATOM_STRING,
+                                               8,
+                                               static_cast<uint32_t>(title.size()),
+                                               title.c_str());
+        AZ_Assert(ValidateXcbResult(xcbCheckResult), "Failed to set window title.");
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void NativeWindowImpl_Linux_xcb::ResizeClientArea(WindowSize clientAreaSize)
+    {
+        const uint32_t values[] = { clientAreaSize.m_width, clientAreaSize.m_height };
+
+        xcb_configure_window(m_xcbConnection, m_xcbWindow, XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT, values);
+        
+        m_width = clientAreaSize.m_width;
+        m_height = clientAreaSize.m_height;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    uint32_t NativeWindowImpl_Linux_xcb::GetDisplayRefreshRate() const
+    {
+        //Using 60 for now until proper support is added
```

Comment:
nit: would be good to add a GHI/jira number for this issue so its not lost.

Reply:
Done